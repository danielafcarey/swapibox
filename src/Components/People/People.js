import React from 'react';
import PropTypes from 'prop-types';

import Card from '../Card/Card.js';

const People = ({ cardData }) => {
  const peopleCards = cardData.map(card => {
    return (
      <Card 
        section='people'
        { ...card }
      />
    )
  })

  return (
    <div className="People">
      { peopleCards }
    </div>
  );
}

People.propTypes = {
  cardData: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default People;
