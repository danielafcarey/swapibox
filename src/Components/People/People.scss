.People{
  .Card {
    background: rgba(43, 43, 43, 0.65) url('../../images/people.png') no-repeat center;
  }
}
