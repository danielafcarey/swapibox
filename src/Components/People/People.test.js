import React from 'react';
import { shallow } from 'enzyme';
import People from './People';

describe('People', () => {

  it('matches the snapshot with one card', () => {
    const mockCardData = [{
      id: 1,
      favorite: false,
      name: 'Daniela',
      homeworld: 'Earth',
      homeworldPop: 7000000000,
      species: ['pupper']
    }];
    const wrapper = shallow(
      <People 
        cardData={ mockCardData }
        toggleFavorite={ jest.fn() }
      /> 
    );

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with cards', () => {
    const mockCardData = [{
      id: 1,
      favorite: false,
      name: 'Daniela',
      homeworld: 'Earth',
      homeworldPop: 7000000000,
      species: ['pupper']
    },
      {
        id: 1,
        favorite: false,
        name: 'Daniela',
        homeworld: 'Earth',
        homeworldPop: 7000000000,
        species: ['pupper']
      }];
    const wrapper = shallow(
      <People 
        cardData={ mockCardData }
        toggleFavorite={ jest.fn() }
      /> 
    );

    expect(wrapper).toMatchSnapshot();
  })

});

