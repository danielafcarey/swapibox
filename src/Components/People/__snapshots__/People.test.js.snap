// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People matches the snapshot with cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    cardData={
        Array [
            Object {
              "favorite": false,
              "homeworld": "Earth",
              "homeworldPop": 7000000000,
              "id": "id",
              "name": "Daniela",
              "species": Array [
                "pupper",
              ],
            },
            Object {
              "favorite": false,
              "homeworld": "Earth",
              "homeworldPop": 7000000000,
              "id": "id",
              "name": "Daniela",
              "species": Array [
                "pupper",
              ],
            },
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={false}
          id="id"
          item1="Homeworld: Earth"
          item2="Population: 7000000000"
          item3="Species: pupper"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
        <Card
          favorite={false}
          id="id"
          item1="Homeworld: Earth"
          item2="Population: 7000000000"
          item3="Species: pupper"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
      ],
      "className": "card-container People",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "id",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": "id",
          "item1": "Homeworld: Earth",
          "item2": "Population: 7000000000",
          "item3": "Species: pupper",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "id",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": "id",
          "item1": "Homeworld: Earth",
          "item2": "Population: 7000000000",
          "item3": "Species: pupper",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={false}
            id="id"
            item1="Homeworld: Earth"
            item2="Population: 7000000000"
            item3="Species: pupper"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
          <Card
            favorite={false}
            id="id"
            item1="Homeworld: Earth"
            item2="Population: 7000000000"
            item3="Species: pupper"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
        ],
        "className": "card-container People",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "id",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": "id",
            "item1": "Homeworld: Earth",
            "item2": "Population: 7000000000",
            "item3": "Species: pupper",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "id",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": "id",
            "item1": "Homeworld: Earth",
            "item2": "Population: 7000000000",
            "item3": "Species: pupper",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People matches the snapshot with one card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    cardData={
        Array [
            Object {
              "favorite": false,
              "homeworld": "Earth",
              "homeworldPop": 7000000000,
              "id": "id",
              "name": "Daniela",
              "species": Array [
                "pupper",
              ],
            },
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={false}
          id="id"
          item1="Homeworld: Earth"
          item2="Population: 7000000000"
          item3="Species: pupper"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
      ],
      "className": "card-container People",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "id",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": "id",
          "item1": "Homeworld: Earth",
          "item2": "Population: 7000000000",
          "item3": "Species: pupper",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={false}
            id="id"
            item1="Homeworld: Earth"
            item2="Population: 7000000000"
            item3="Species: pupper"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
        ],
        "className": "card-container People",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "id",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": "id",
            "item1": "Homeworld: Earth",
            "item2": "Population: 7000000000",
            "item3": "Species: pupper",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
